### PR TITLE
Add position roles

### LINKED ISSUE #94: Top counter has a role
# Feature request

Create a way to add a flow and action to the top counter / score in a server. I’m attempting to have a role automatically assigned to the top counter, and removed from anyone else who was top counter before.

This would be great for competition within a server to maintain the highest count. And just easier on the admins for not having to add/remove a top role every day.

Suggested by Fuego#0003

### PR DESCRIPTION
**Pull request by promise**
Fixes #94

### CODE REVIEW

File: src/commands/chatInput/positionroles/set.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      type: ApplicationCommandOptionType.Role,
+      name: "role",
+      description: "The role you want to give out",
+      required: true,
+    },
+    {
+      type: ApplicationCommandOptionType.Integer,
+      name: "position",
+      description: "The scoreboard position you want to give the role out at",
+      required: true,
```
Comment (Ta1al) [inline src/commands/chatInput/positionroles/set.ts:22]: `minValue` & `maxValue` can be added here

File: src/commands/chatInput/positionroles/set.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    const me = await interaction.guild.members.fetchMe({ force: false }).catch(() => null);
+    if (me && role.comparePositionTo(me.roles.highest) > 0) {
+      return void interaction.reply({
+        content: "❌ The role must be lower than the bot's highest role.",
+        ephemeral: true,
+      });
+    }
+
+    if (position < 1 || position > limits.positionRole.max) {
```
Comment (Ta1al) [inline src/commands/chatInput/positionroles/set.ts]: This check isn't needed if `minValue` & `maxValue` in the interaction option

**Review by Ta1al — CHANGES_REQUESTED** [review]
There is currently no way to remove the position roles.

**Comment by promise** [comment]
will work on removing position roles soon, but pls review current changes

**Review by Ta1al — APPROVED** [review]

**Review by djohts — APPROVED** [review]

File: src/commands/chatInput/positionroles/list.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+          callback(cancel) {
+            return void cancel.update({
+              content: "💨 Cancelled deletion.",
+              components: [],
+            });
+          },
+        });
+
+        return void select.reply({
+          content: `💢 Are you sure you want to delete position roles for position${positions.length === 1 ? "" : "s"} ${formatListToHuman(positions)}?`,
```
Comment (Ta1al) [inline src/commands/chatInput/positionroles/list.ts]: For grammar's sake, at a plural check for **roles** as well.

**Review by Ta1al — CHANGES_REQUESTED** [review]

**Review by Ta1al — APPROVED** [review]

**Pull request merged**